test/schematic_test.exs

```diff
@@ -105,6 +178,89 @@ defmodule SchematicTest do
       assert {:ok, %{"foo" => "hi there!"}} == unify(schematic, input)
     end
 
+    property "map/1 using schematic options" do
+      check all(
+              schematics <- StreamData.list_of(Generators.schematic(), min_length: 1),
+              input <-
+                StreamData.map_of(
+                  StreamData.binary(),
+                  Enum.map(schematics, &Generators.from_schematic/1) |> StreamData.one_of()
+                )
+            ) do
+        schematic = map(key_schematic: str(), value_schematic: oneof(schematics))
+
+        assert {:ok, input} == unify(schematic, input)
+      end
+    end
+
+    property "map/1 using blueprint" do
+      check all(
+              {blueprint, input} <-
+                StreamData.bind(
+                  StreamData.map_of(StreamData.binary(), Generators.schematic(), min_length: 1),
```

Comment:
map schematics can technically have any value as a key in the blueprint, so we would want to expand this generator to reflect that.

We'll also want it to generate map blue prints that reflect the key transformation capabilities: https://hexdocs.pm/schematic/Schematic.html#map/1-transforming-keys